assets/src/components/dummyDetourPage.tsx

```diff
@@ -1,8 +1,37 @@
-import React, { useEffect, useState } from "react"
+import React, { useEffect, useMemo, useState } from "react"
 import { fetchRoutePatterns } from "../api"
 import { RoutePattern } from "../schedule"
 import { DiversionPage } from "./detours/diversionPage"
 import { useRoute } from "../contexts/routesContext"
+import { createDetourMachine } from "../models/createDetourMachine"
+import { AnyActorLogic, Snapshot, createActor } from "xstate"
+import { reload } from "../models/browser"
+import { isOk, Err, Ok, Result } from "../util/result"
+
+/**
+ * Checks if the provided object, {@linkcode snapshot}, is a "valid" snapshot
+ * for the provided state machine {@linkcode actorLogic}.
+ */
+const isValidSnapshot = (
```

Comment:
I'm wondering if this should be a util function in the `useDetour` hook file?

Reply:
I wasn't sure where to put it because I made it a bit more generic than "just" the `createDetourMachine` (see the `actorLogic` arg)

Given that, do you have any other thoughts on _where_ it should go? I'm open to it going in `useDetour` for now anyway

Reply:
I decided to move it into it's own file https://github.com/mbta/skate/compare/af7ed658055f2747717e946df684ca5b6f88343b..5c74ac9ed78cf2434377aebbf020ced67c8b92eb

Reply:
Oh yes, that's great. Love that it's generic and reusable. Thanks!